modules/shared/src/main/scala/io/constellationnetwork/env/JarSignature.scala

```diff
@@ -0,0 +1,42 @@
+package io.constellationnetwork.env
+
+import java.security.MessageDigest
+
+import cats.effect.Sync
+import cats.implicits.{catsSyntaxFlatMapOps, toFunctorOps}
+
+import io.constellationnetwork.security.hash.Hash
+import io.constellationnetwork.security.hex.Hex
+
+import fs2.Chunk
+import fs2.io.file.{Files, Path}
+
+object JarSignature {
+
+  def jarHash[F[_]: Sync: Files]: F[Hash] = {
+    val jarPath = JarSignature.getClass.getProtectionDomain.getCodeSource.getLocation.toURI
+    digestOf(Files[F].readAll(Path(jarPath.getPath)))
+  }
+
+  def digestOf[F[_]: Sync: Files](bytes: fs2.Stream[F, Byte]): F[Hash] = {
+    val chunkSize = 524288
+
+    digestInstance >>= { digest =>
+      bytes
+        .chunkN(chunkSize)
+        .evalMap(updateDigest(digest, _))
+        .compile
+        .drain >> toHash(digest)
+    }
+  }
+
+  private def digestInstance[F[_]: Sync] =
+    Sync[F].delay(MessageDigest.getInstance("SHA-256"))
+
+  private def updateDigest[F[_]: Sync](digest: MessageDigest, chunk: Chunk[Byte]) =
+    Sync[F].delay(digest.update(chunk.toArray))
+
+  private def toHash[F[_]: Sync](digest: MessageDigest) =
+    Sync[F].delay(digest.digest()).map(d => Hash(Hex.fromBytes(d).value))
```

Comment:
`.map(Hex.fromBytes).map(_value).map(Hash)`

Reply:
I like pointfree and in this case doesn't make a difference, so I'll change it.
That said,  Hash and Hex are newtypes, so Hash(_) and  hex.value are just casts (essentially no-ops ) and for performance critical code, I think we should avoid sending empty IO.Map objects to the main effect loop.